index.js

```diff
@@ -1,4 +1,83 @@
 //  in this file create an express application - use the middle-ware built into express
-//  to serve up static files from the public directory (index.html and client.js - you 
+//  to serve up static files from the public directory (index.html and client.js - you
 //  can also serve up css files from the public directory this way if you want)
-//  you need to support a '/trucks' endpoint, and a dynamic route for '/trucks/:name'
\ No newline at end of file
+//  you need to support a "/trucks" endpoint, and a dynamic route for "/trucks/:name"
+
+var express = require("express");
+var trucks = require("./trucks");
+var app = express();
+
+var serveStatic = express.static("public");
+app.use(serveStatic);
+ 
+app.get("/trucks", function(request, response) {
+    var truckStr = "";
+    var truckArray = trucks.getTrucks();
+    for (var i = 0; i < truckArray.length; i++) {
+        truckStr += truckArray[i].name + "<br>";
+    }
+
+    response.send(truckStr);
+});
+
+app.param("day", function(request, response, next) {
+    var day = request.params.day;
+    var truck = trucks.filterTrucksByDay(day);
+    request.truck = truck;
+    next();
+});
+
+app.get("/trucks/day/:day", function(request, response) {
+    var truckStr = "";
+    var truckArray = request.truck;
+    for (var i = 0; i < truckArray.length; i++) {
+        truckStr += truckArray[i].name + "<br>";
+    }
+
+    response.send(truckStr);
+});
+
+app.param("name", function(request, response, next) {
+    var name = request.params.name.toLowerCase();
+    var truck = trucks.getTruck(name);
```

Comment:
When you create a generalized params handler like this, you typically want to do the work of retrieving things using it in individual handlers. For example, if you had two different routes that both use a `name` parameter, you would cache the name here, then use it to do work in the handler, in this case call `getTruck` on your trucks module.